Source/WebCore/style/Styleable.cpp

```diff
@@ -735,8 +735,36 @@ void Styleable::updateCSSTransitions(const RenderStyle& currentStyle, const Rend
     if (currentStyle.display() == DisplayType::None)
         return;
 
-    // In case this element is newly getting a "display: none" we need to cancel all of its transitions and disregard new ones.
-    if (currentStyle.hasTransitions() && currentStyle.display() != DisplayType::None && newStyle.display() == DisplayType::None) {
+    auto transitionsDisplay = [](const RenderStyle& style) {
+        if (!style.hasTransitions())
+            return false;
+
+        for (auto& transition : *style.transitions()) {
+            auto transitionProperty = transition->property();
+            switch (transitionProperty.mode) {
+            case Animation::TransitionMode::All:
+                if (transition->allowsDiscreteTransitions())
+                    return true;
```

Comment:
You could probably do:

```
if (!transition->allowsDiscreteTransitions())
    continue;
```

at the beginning of the loop and remove the individual checks, which might save a bit of work. No strong opinion though.